modules/contenta_enhancements/contenta_enhancements.module

```diff
@@ -21,3 +48,300 @@ function contenta_enhancements_entity_base_field_info_alter(&$fields, EntityType
+    $items['contenta_content_models'] = $empty_item_cacheability;
+  }
+
+  if ($current_user->hasPermission('access openapi api docs')) {
+    $items['contenta_api'] = [
+      '#type' => 'toolbar_item',
+      'tab' => [
+        '#type' => 'link',
+        '#title' => t('API'),
+        '#url' => Url::fromRoute('contenta_enhancements.api'),
+        '#options' => [
+          'set_active_class' => TRUE,
+        ],
+        '#attributes' => [
+          'class' => ['toolbar-icon', 'toolbar-icon-system-modules-list'],
+        ],
+      ],
+      '#weight' => -25,
+    ];
+  }
+  else {
+    $items['contenta_api'] = $empty_item_cacheability;
+  }
+
+  if ($current_user->hasPermission('administer users') || $current_user->hasPermission('administer simple_oauth entities')) {
+    $items['contenta_data_access'] = [
+      '#type' => 'toolbar_item',
+      'tab' => [
+        '#type' => 'link',
+        '#options' => [
+          'set_active_class' => TRUE,
+        ],
+        '#title' => t('Access control'),
+        '#url' => $current_user->hasPermission('administer users')
+          ? Url::fromRoute('entity.user.collection')
+          : Url::fromRoute('entity.oauth2_token.collection'),
+        '#attributes' => [
+          'class' => ['toolbar-icon', 'toolbar-icon-entity-user-collection'],
+        ],
+      ],
+      '#weight' => -20,
+    ];
+  }
+  else {
+    $items['contenta_data_access'] = $empty_item_cacheability;
+  }
+
+  if ($current_user->hasPermission('access site reports')) {
+    $menu_tree = \Drupal::service('toolbar.menu_tree');
+    $parameters = new MenuTreeParameters();
+    $parameters->setMinDepth(1)->setMaxDepth(3)->onlyEnabledLinks();
+    $parameters->setRoot('system.admin_reports');
+    $tree = $menu_tree->load('administration', $parameters);
+    $manipulators = array(
+      array('callable' => 'menu.default_tree_manipulators:checkAccess'),
+      array('callable' => 'menu.default_tree_manipulators:generateIndexAndSort'),
+      array('callable' => 'toolbar_menu_navigation_links'),
+    );
+    $tree = $menu_tree->transform($tree, $manipulators);
+    $renderable_reports = $menu_tree->build($tree);
+    $items['reports'] = [
+      '#type' => 'toolbar_item',
+      'tab' => [
+        '#type' => 'link',
+        '#title' => t('Reports'),
+        '#url' => Url::fromRoute('system.admin_reports'),
+        '#attributes' => [
+          'title' => t('Reports'),
+          'class' => [
+            'toolbar-icon',
+            'toolbar-icon-system-admin-reports',
+          ],
+        ],
+      ],
+      'tray' => [
+        '#heading' => t('Reports'),
+        '#wrapper_attributes' => [],
+        'menu' => $renderable_reports,
+      ],
+      '#weight' => -5,
+    ];
+  }
+  else {
+    $items['reports'] = $empty_item_cacheability;
+  }
+
+  $items['administration']['tab']['#title'] = t('Advanced');
+  $items['administration']['#weight'] = 30;
+  if (!$current_user->hasPermission('access administration pages')) {
```

Comment:
Probably, but they seem to want to have the empty item for cacheability metadata. Surely @wimleers has some input here.